### ISSUE TITLE
How to have compiled files in a sub-folder

### ISSUE THREAD

**Issue by phillip-haydon** [angular/angular-cli#3932]
> Please provide us with the following information:
> ---------------------------------------------------------------

### OS?
> Windows 7, 8 or 10. Linux (which distribution). Mac OSX (Yosemite? El Capitan?)

All

### Versions.
> Please run `ng --version`. If there's nothing outputted, please run in a Terminal: `node --version` and paste the result here:

angular-cli: 1.0.0-beta.19-3
node: 6.3.1
os: win32 x64

### Repro steps.
> Was this an app that wasn't created using the CLI? What change did you do on your code? etc.

n/a

### The log given by the failure.
> Normally this include a stack trace and some more information.

n/a

### Mention any other details that might be useful.

I'm hosting a site in AWS S3, and the site, and different API's are passed through Gateway API. 

In order to have routing working in Angular I have to have a proxy resource which returns the index.html from S3, then have all assets loaded from a sub-directory from /assets/....

If I move the files manually I get:
```
Uncaught TypeError: e.map is not a function
    at t.parameters (https://app.*site*.net/assets/main.80e8f7b5792f4a5fc6e9.bundle.js:1346:663)
    at e.parameters (https://app.*site*.net/assets/main.80e8f7b5792f4a5fc6e9.bundle.js:1353:502)
    at t.getDependenciesMetadata (https://app.*site*.net/assets/main.80e8f7b5792f4a5fc6e9.bundle.js:638:9225)
    at t.getTypeMetadata (https://app.*site*.net/assets/main.80e8f7b5792f4a5fc6e9.bundle.js:638:8543)
    at t.getDirectiveMetadata (https://app.*site*.net/assets/main.80e8f7b5792f4a5fc6e9.bundle.js:638:3577)
    at https://app.*site*.net/assets/main.80e8f7b5792f4a5fc6e9.bundle.js:638:4705
    at Array.forEach (native)
    at t.getNgModuleMetadata (https://app.*site*.net/assets/main.80e8f7b5792f4a5fc6e9.bundle.js:638:4564)
    at https://app.*site*.net/assets/main.80e8f7b5792f4a5fc6e9.bundle.js:638:4373
    at Array.forEach (native)
```

I could solve this by introducing Cloudfront for the assets then setting up 404 behavior to return the /index.html, but then I can no longer version off the assets in S3.

Basically without the ability to have sub-folders for the assets, angular-cli becomes useless :(

I don't want to have to drop the CLI as it's super handy and makes Angular more enjoyable. 

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by filipesilva** [comment]
Fixed by https://github.com/angular/angular-cli/pull/3285.
